### PR TITLE
Fixed '--libraries' and '--library' flags description in compile command

### PR DESCRIPTION
**Pull request by cmaglie**
**Please check if the PR fulfills these requirements**

- [x] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-cli/pulls)
      before creating one)
- [x] The PR follows
      [our contributing guidelines](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#pull-requests)
- [] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] `UPGRADING.md` has been updated with a migration guide (for breaking changes)

**What kind of change does this PR introduce?**
Fixed '--libraries' and '--library' flags description in compile command

**What is the current behavior?**
the description obtained from `arduino-cli compile -h` is:
```
      --libraries strings            List of custom libraries dir paths separated by commas. Or can be used multiple times for multiple libraries dir paths.
      --library strings              List of paths to libraries root folders. Libraries set this way have top priority in case of conflicts. Can be used multiple times for different libraries.
```

and is wrong (well, at least is not very clear...)

* **What is the new behavior?**
```
      --libraries strings            Path to a collection of libraries. Contained libraries override the ones found in the default libraries folder. Can be used multiple times or entries can be comma separated.
      --library strings              Path to a single library’s root folder. Overrides other occurrences of the library found in default libraries folder or passed in via --libraries. Can be used multiple times or entries can be comma separated.
```

**Does this PR introduce a breaking change, and is [titled accordingly](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#breaking)?**
No

### CODE REVIEW

**Review by per1234 — COMMENTED** [review]
I'm wondering whether it is wise to attempt an explanation of library priority here. Location is [only one of the factors](https://arduino.github.io/arduino-cli/dev/sketch-build-process/#location-priority).

File: cli/compile/compile.go [hunk 1]
```diff
@@ -116,9 +116,9 @@ func NewCommand() *cobra.Command {
 	compileCommand.Flags().BoolVarP(&verify, "verify", "t", false, tr("Verify uploaded binary after the upload."))
 	compileCommand.Flags().StringVar(&vidPid, "vid-pid", "", tr("When specified, VID/PID specific build properties are used, if board supports them."))
 	compileCommand.Flags().StringSliceVar(&library, "library", []string{},
-		tr("List of paths to libraries root folders. Libraries set this way have top priority in case of conflicts. Can be used multiple times for different libraries."))
+		tr("Path to a single library’s root folder. Overrides other occurrences of the library found in default libraries folder or passed in via --libraries. Can be used multiple times or entries can be comma separated."))
```
Comment (per1234) [inline cli/compile/compile.go:119]:
```suggestion
		tr("Path to a single library’s root folder. Can be used multiple times or entries can be comma separated."))
```

File: cli/compile/compile.go [hunk 2]
```diff
@@ -116,9 +116,9 @@ func NewCommand() *cobra.Command {
...
 	compileCommand.Flags().StringSliceVar(&libraries, "libraries", []string{},
-		tr("List of custom libraries dir paths separated by commas. Or can be used multiple times for multiple libraries dir paths."))
+		tr("Path to a collection of libraries. Contained libraries override the ones found in the default libraries folder. Can be used multiple times or entries can be comma separated."))
```
Comment (per1234) [inline cli/compile/compile.go:121]:
```suggestion
		tr("Path to a collection of libraries. Can be used multiple times or entries can be comma separated."))
```

**Review by per1234 — CHANGES_REQUESTED** [review]
I may not have communicated clearly. I was not suggesting that we should add to it. I was suggesting that we remove the incomplete information about library priority.

**Comment by cmaglie** [comment]
That makes sense. I've also changed the description of the gRPC API.

**Pull request merged**